src/player.cpp

```diff
@@ -4761,6 +4761,17 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
                                 source->disp_name().c_str());
             }
         }
+
+        // Melee damage causing lacerations through filthy clothing may result in infection
+        if( d.type_damage( DT_CUT ) || d.type_damage( DT_STAB ) ) {
+            bool filthy = std::any_of( worn.begin(), worn.end(), [bp]( const item& e ) {
+                return e.covers( bp ) && e.is_filthy();
+            } );
+            if( filthy && one_in( 2 ) ) {
```

Comment:
One in 2 is a lot, but cutting damage is very rare.

I'd suggest some other measure of infection, preferably allowing bashing and filthy clothing stacking.

I'm thinking something like:
```
infection_chance = d.damage_type( DT_BASH ) + ( d.damage_type( DT_CUT ) + d.damage_type( DT_STAB ) ) * 4;
bool filthy = std::any_of( worn.begin(), worn.end(), [bp]( const item& e ) {
    return e.covers( bp ) && e.is_filthy() && x_in_y( infection_chance, 100 );
} );
```

That way: cutting damage would still matter, stabbing damage would still matter, bashing damage would actually matter too, total damage amount would matter (meaning grazing wounds wouldn't be as infectious), multiple dirty items would matter.

Could also add a coverage scaling to the infectiousness. Wearing a filthy backpack would be less likely to get someone infected than wearing a hoodie guaranteeing that almost all upper body is covered in zombie sludge.